src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts

```diff
@@ -299,38 +309,62 @@ export class ExtensionsViewlet extends ViewContainerViewlet implements IExtensio
 				this.defaultRecommendedExtensionsContextKey.set(!this.configurationService.getValue<boolean>(ShowRecommendationsOnlyOnDemandKey));
 			}
 		}, this, this.disposables);
+
+		modes.SuggestRegistry.register({ scheme: 'extensions', pattern: '**/searchinput', hasAccessToAllModels: true }, {
+			triggerCharacters: ['@'],
+			provideCompletionItems: (model: ITextModel, position: Position, _context: modes.SuggestContext) => {
+				return {
+					suggestions: this.autoComplete(model.getValue(), position.column).map(item => (
+						{
+							label: item.fullText,
+							insertText: item.fullText,
+							overwriteBefore: item.overwrite,
+							sortText: item.fullText.indexOf(':') === -1 ? 'a' : item.fullText.indexOf('sort') !== -1 ? 'b' : 'c', // things with no colon; sorts; everything else
+							type: <modes.SuggestionType>'keyword'
+						}))
+				};
+			}
+		});
 	}
 
 	create(parent: HTMLElement): TPromise<void> {
 		addClass(parent, 'extensions-viewlet');
 		this.root = parent;
 
 		const header = append(this.root, $('.header'));
-
-		this.searchBox = append(header, $<HTMLInputElement>('input.search-box'));
-		this.searchBox.placeholder = localize('searchExtensions', "Search Extensions in Marketplace");
-		this.disposables.push(addStandardDisposableListener(this.searchBox, EventType.FOCUS, () => addClass(this.searchBox, 'synthetic-focus')));
-		this.disposables.push(addStandardDisposableListener(this.searchBox, EventType.BLUR, () => removeClass(this.searchBox, 'synthetic-focus')));
+		this.monacoStyleContainer = append(header, $('.monaco-container'));
+		this.searchBox = this.instantiationService.createInstance(CodeEditorWidget, this.monacoStyleContainer, SEARCH_INPUT_OPTIONS, {});
+		this.placeholderText = append(this.monacoStyleContainer, $('.search-placeholder', null, 'Search Extensions in Marketplace'));
 
 		this.extensionsBox = append(this.root, $('.extensions'));
 
-		const onKeyDown = chain(domEvent(this.searchBox, 'keydown'))
-			.map(e => new StandardKeyboardEvent(e));
-		onKeyDown.filter(e => e.keyCode === KeyCode.Escape).on(this.onEscape, this, this.disposables);
+		this.searchBox.setModel(this.modelService.createModel('', null, uri.parse('extensions:searchinput'), true));
+
+		this.disposables.push(this.searchBox.onDidFocusEditorText(() => addClass(this.monacoStyleContainer, 'synthetic-focus')));
+		this.disposables.push(this.searchBox.onDidBlurEditorText(() => removeClass(this.monacoStyleContainer, 'synthetic-focus')));
 
-		const onKeyDownForList = onKeyDown.filter(() => this.count() > 0);
-		onKeyDownForList.filter(e => e.keyCode === KeyCode.Enter).on(this.onEnter, this, this.disposables);
+		const onKeyDownMonaco = chain(this.searchBox.onKeyDown);
+		onKeyDownMonaco.filter(e => e.keyCode === KeyCode.Tab).on(e => e.stopPropagation(), this, this.disposables);
```

Comment:
Pressing tab when the suggest widget is open should result in the selected suggestion being accepted. But this doesnt happen due to the above.

Reply:
For this and below, I don't know of a good way to enable keys based on the suggest widget being open. the repl does something with contextkeys that's moderately complicated and doesn't actually work all that well.

Reply:
I'll look into what we do in the core editor

Reply:
So core editor doesn't actually help much here I think, because in core editor they don't need to disable tabbing inside the editor. Unfortunately I cant find a way to get a trigger when the dropdown closes (the dropdown can close on click in addition to tab/esc/enter; we can listen to the keys, but not the click)